server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/builder/BaseStarTreeBuilder.java

```diff
@@ -325,25 +593,26 @@ protected StarTreeDocument reduceSegmentStarTreeDocuments(
     /**
      * Safely converts the metric value of object type to long.
      *
-     * @param metric value of the metric
+     * @param metric                value of the metric
+     * @param idempotentMetricValue
      * @return converted metric value to long
      */
-    private static long getLong(Object metric) {
+    private static long getLong(Object metric, long idempotentMetricValue) {
 
-        Long metricValue = null;
+        long metricValue;
         try {
             if (metric instanceof Long) {
                 metricValue = (long) metric;
             } else if (metric != null) {
-                metricValue = Long.valueOf(String.valueOf(metric));
+                metricValue = Long.parseLong(String.valueOf(metric));
+            } else {
+                logger.debug("metric value is null, returning idempotent metric value for the aggregator");
```

Comment:
nice change.

Reply:
We can remove debug logs for this , for each document , for each identity metric this will be printed.